trenchboot/02-with-drtm.robot

```diff
@@ -0,0 +1,133 @@
+*** Settings ***
+Library             OperatingSystem
+Library             Telnet    timeout=20 seconds    connection_timeout=120 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+Resource            ../sonoff-rest-api/sonoff-api.robot
+Resource            ../rtectrl-rest-api/rtectrl.robot
+Resource            ../variables.robot
+Resource            ../keywords.robot
+Resource            ../keys.robot
+Resource            ../lib/trenchboot.robot
+Resource            ../lib/tpm.robot
+
+Suite Setup         TrenchBoot Suite Setup
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+# WTD -- WiTh Drtm
+#
+# These tests verify sanity of the platform when DRTM is enabled.
+#
+# Supported TPM: 1.2 or 2.0 (SHA1 and/or SHA256 PCR banks)
+
+
+*** Test Cases ***
+WTD001.001 All cores are up
+    [Documentation]    Verifies that all CPUs are online with DRTM.
+    ${online}=    Execute Command In Terminal
+    ...    cat /sys/devices/system/cpu/online
+    ${possible}=    Execute Command In Terminal
+    ...    cat /sys/devices/system/cpu/possible
+    Should Be Equal As Strings    ${online}    ${possible}
+    ...    Not all cores are up
+
+WTD002.001 DRTM PCRs are updated with TB
+    [Documentation]    Checks that DRTM changes 17-22 PCRs as it should.
+
+    ${pcr_hashes}=    Get PCRs State From Linux    1[7-8]
+    FOR    ${pcr_hash}    IN    @{pcr_hashes}
+        ${pcr}    ${hash}=    Split String    ${pcr_hash}    separator=:
+        ${unique_values_str}=    Evaluate    ''.join(set("${hash}"))
+        Should Not Be Equal    ${unique_values_str}    F    msg=${pcr}
+        Should Not Be Equal    ${unique_values_str}    0    msg=${pcr}
+    END
+
+    ${pcr_hashes}=    Get PCRs State From Linux    19|2[0-2]
+    FOR    ${pcr_hash}    IN    @{pcr_hashes}
+        ${pcr}    ${hash}=    Split String    ${pcr_hash}    separator=:
+        ${unique_values_str}=    Evaluate    ''.join(set("${hash}"))
+        Should Not Be Equal    ${unique_values_str}    F    msg=${pcr}
+        Should Not Be Equal    ${unique_values_str}    0    msg=${pcr}
+    END
+
+WTD003.001 Event log exists
+    [Documentation]    Verifies that the event log is present with DRTM.
+    ${present}=    Execute Command In Terminal
+    ...    test -f /sys/kernel/security/tpm0/binary_bios_measurements && echo y
+    Should Be Equal As Strings    ${present}    y
+
+WTD004.001 Event log has DRTM entries (AMD)
```

Comment:
SL event log will be exposed separately to SRTM one, see https://github.com/TrenchBoot/linux/commit/792e72143ba50600f096c3856ba8eaff2e7eb758#diff-6027baf7ab71879448c4e249e4b1f3fa2689a6c1d7113a564b1605ae9850a995R200 and directory structure leading to it. The rest of this test will have to be modified accordingly, or better create separate tests for existence and validity of log.

Reply:
https://github.com/Dasharo/open-source-firmware-validation/compare/5163e1a22995279403dcc7f277ea309cd4ac79b3..f3c14936b5b14706e5efa3208f4544fe2ee65cd3#diff-7e5ab2b2ca7ecfb07ecd58004916bea6a39f406b2ccb9c3cf7a9ffdd90218011R56-R76

> SL event log will be exposed separately to SRTM one

Does this mean that https://github.com/Dasharo/open-source-firmware-validation/compare/5163e1a22995279403dcc7f277ea309cd4ac79b3..f3c14936b5b14706e5efa3208f4544fe2ee65cd3#diff-7e5ab2b2ca7ecfb07ecd58004916bea6a39f406b2ccb9c3cf7a9ffdd90218011R100-R103 needs to check that separate log as well?  Logs are independent and don't need to be combined for replaying, right?

Reply:
Right, SRTM shouldn't touch PCRs 17-22 and vice versa. Both should match, but to distinguish between failure in SRTM and DRTM, these should be in separate tests. We still want to check that DRTM is good even with bad SRTM event log caused by buggy firmware (Dasharo or not).

Reply:
https://github.com/Dasharo/open-source-firmware-validation/compare/f3c14936b5b14706e5efa3208f4544fe2ee65cd3..209c33b041b37bfce1d54751cbb4648c9ffe0550